src/util/mpir_hw_topo.c

```diff
@@ -459,6 +459,35 @@ MPIR_Node_obj MPIR_Node_get_parent_obj(MPIR_Node_obj obj)
     return ret;
 }
 
+MPIR_Node_obj MPIR_Node_get_obj_by_type(MPIR_Node_obj_type obj_type, int obj_index)
+{
+    MPIR_Node_obj ret = NULL;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIR_NODE_GET_OBJ_BY_TYPE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIR_NODE_GET_OBJ_BY_TYPE);
+
+    if (!hw_topo.bindset_is_valid)
+        goto fn_exit;
+
+#ifdef HAVE_HWLOC
+    hwloc_obj_type_t hw_obj_type = convert_mpir_node_obj_type_to_hw(obj_type);
+    hwloc_obj_t hw_obj = hwloc_get_obj_by_type(hw_topo.hwloc_topology, hw_obj_type, obj_index);
+    if (obj_type == MPIR_NODE_OBJ_TYPE__DDR) {
+        if (!hw_obj->subtype)
+            ret = hw_obj;
+    } else if (obj_type == MPIR_NODE_OBJ_TYPE__HBM) {
+        if (hw_obj->subtype)
```

Comment:
We do this again and again. How about a macro, e.g. `HW_OBJ_IS_TYPE(obj, type)`?